Description: Fix potential loss of data if filesystem is full, take patch from upstream
 svn revision 3044. (LP: #1271104)
Author: Wouter Wijngaards
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1271104
Bug: https://www.nlnetlabs.nl/bugs-script/show_bug.cgi?id=547

--- unbound-1.4.21.orig/validator/autotrust.c
+++ unbound-1.4.21/validator/autotrust.c
@@ -999,8 +999,13 @@ void autr_write_file(struct module_env*
 		log_err("could not completely write: %s", fname);
 		return;
 	}
+	if(fclose(out) != 0) {
+		log_err("could not complete write: %s: %s",
+			fname, strerror(errno));
+		unlink(tempf);
+		return;
+	}
 	/* success; overwrite actual file */
-	fclose(out);
 	verbose(VERB_ALGO, "autotrust: replaced %s", fname);
 #ifdef UB_ON_WINDOWS
 	(void)unlink(fname); /* windows does not replace file with rename() */
